kinesis/src/main/scala/akka/stream/alpakka/kinesis/scaladsl/KinesisFlow.scala

```diff
@@ -7,29 +7,38 @@ package akka.stream.alpakka.kinesis.scaladsl
 import java.nio.ByteBuffer
 
 import akka.NotUsed
+import akka.dispatch.ExecutionContexts.sameThreadExecutionContext
 import akka.stream.ThrottleMode
 import akka.stream.alpakka.kinesis.KinesisFlowSettings
-import akka.stream.alpakka.kinesis.impl.KinesisFlowStage
+import akka.stream.alpakka.kinesis.KinesisErrors.FailurePublishingRecords
 import akka.stream.scaladsl.Flow
 import akka.util.ByteString
-import com.amazonaws.services.kinesis.AmazonKinesisAsync
-import com.amazonaws.services.kinesis.model.{PutRecordsRequestEntry, PutRecordsResultEntry}
+import software.amazon.awssdk.core.SdkBytes
+import software.amazon.awssdk.services.kinesis.KinesisAsyncClient
+import software.amazon.awssdk.services.kinesis.model.{
+  PutRecordsRequest,
+  PutRecordsRequestEntry,
+  PutRecordsResponse,
+  PutRecordsResultEntry
+}
 
+import scala.collection.JavaConverters._
 import scala.collection.immutable.Queue
 import scala.concurrent.duration._
+import scala.compat.java8.FutureConverters._
 
 object KinesisFlow {
 
   def apply(streamName: String, settings: KinesisFlowSettings = KinesisFlowSettings.Defaults)(
-      implicit kinesisClient: AmazonKinesisAsync
+      implicit kinesisClient: KinesisAsyncClient
   ): Flow[PutRecordsRequestEntry, PutRecordsResultEntry, NotUsed] =
     Flow[PutRecordsRequestEntry]
       .map((_, ()))
       .via(withUserContext(streamName, settings))
       .map(_._1)
 
   def withUserContext[T](streamName: String, settings: KinesisFlowSettings = KinesisFlowSettings.Defaults)(
-      implicit kinesisClient: AmazonKinesisAsync
+      implicit kinesisClient: KinesisAsyncClient
   ): Flow[(PutRecordsRequestEntry, T), (PutRecordsResultEntry, T), NotUsed] =
```

Comment:
Akka has been extended to have native support for this pattern now.
Please make it return `FlowWithContext[PutRecordsRequestEntry, T, PutRecordsResultEntry, T, NotUsed]`. The simplest solution is to use `FlowWithContext.fromTuples(...)`.
The methods should be renamed to `withContext`/`createWithContext` to align nicely.